bitnami/mongodb/templates/replicaset/headless-svc.yaml

```diff
@@ -22,6 +22,12 @@ spec:
   type: ClusterIP
   clusterIP: None
   publishNotReadyAddresses: true
+  {{- if .Values.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
```

Comment:
Hi, it is specifically the headless service, where I am seeing issues. Since the headless still uses ClusterIP and resolves multiple pods, it will still be able to use the internal IPs for loadbalancing and the sessionAffinity is still relevant. 

I have tested this, by manually editing the service generated by the helm chart, and running an otherwise failing application. The issue that will arise is `Cursor not found`, for larger/longer queries, since a different pod is served after a while.

Reply:
It is also worth noting that the documentation within the Readme implies that this configuration variable will be set, regardless of whether it is run as headless or not. Therefor it could be better to provide the users the option, either way.

Reply:
@fmulero I will let you resolve the conversation, unless you have any additional input?

Reply:
I've been revisiting this with a colleague and we have mixed feelings about the  `sessionAffinity` here.

Theoretically a headless service is like several 'A' entries in your DNS for the same name, there is no load balancing. We use this service to discover the members of the cluster, that's the reason why this service has the property `publishNotReadyAddresses: true`, which makes non ready pods are attached to the service. Generally speaking, If you use this service in your clients, you can face connectivity issues in case of restarts, upgrades or cluster initialization.

We'd like to know a little bit more about your case of use, because we don't fully understand how setting sessionAffinity here could fix the issue you mentioned.